### PR TITLE
SWM-356 Seperated account freeze and token pause

### PR DESCRIPTION
**Pull request by filippetroviccc**

### CODE REVIEW

File: contracts/token/Freezable.sol [hunk 1]
```diff
@@ -1,54 +1,38 @@
 pragma solidity ^0.5.0;
 
 /**
- * @title Freezable contract
- * @dev Base contract providing internal methods for account
- * and to track what accounts are frozen, and if whole token is frozen.
- * Token freezing is shortcut for freezing every account.
+ * @title Freezable account
```
Comment (tomislavr) [inline contracts/token/Freezable.sol]: "@title Freezable account" should be "@title Freezable contract"

File: contracts/token/Pausable.sol [hunk 2]
```diff
@@ -0,0 +1,73 @@
+pragma solidity ^0.5.0;
+
+import "openzeppelin-solidity/contracts/access/roles/PauserRole.sol";
```
Comment (tomislavr) [inline contracts/token/Pausable.sol]: This is not used, can be removed.

File: test/token/SRC20.test.js [hunk 3]
```diff
@@ -279,27 +279,44 @@ contract('SRC20', async function ([_, manager, owner, authority0, authority1, ac
...
       });
+
+      const isAccountFrozen = await this.token.isFrozen(account0, {from: owner});
+      assert.equal(isAccountFrozen, true);
+
     });
 
-    it('should not allow account and token freezing if feature disabled', async function () {
-      await shouldFail.reverting.withMessage(this.token.freezeToken({from: owner}),
+    it('should not allow token freezing if feature disabled', async function () {
```
Comment (tomislavr) [inline test/token/SRC20.test.js]: "token freezing" should be "token pausing".

File: contracts/token/SRC20.sol [hunk 4]
```diff
@@ -325,40 +326,40 @@ contract SRC20 is ISRC20, ISRC20Owned, ISRC20Managed, SRC20Detailed, Featured,
      *
      * @return True if token is frozen.
      */
-    function isTokenFrozen() external view returns (bool) {
-        return _isTokenFrozen();
+    function isTokenPaused() external view returns (bool) {
```
Comment (tomislavr) [inline contracts/token/SRC20.sol]: I don't see anywhere that transfers are forbidden on paused token. If this is to be the case, please add tests that will check for this condition.

File: contracts/token/Freezable.sol (hunk 1 above)
Comment (filippetroviccc) [reply contracts/token/Freezable.sol]: @title Freezable account contract maybe?

File: contracts/token/SRC20.sol (hunk 4 above)
Comment (filippetroviccc) [reply contracts/token/SRC20.sol]: @tomislavr The check is already their.
https://github.com/swarmfund/swarm-compliant-contract/blob/6497f3a1fa2416bc7637297a7a369e7c5fad9fae/contracts/token/SRC20.sol#L467

Comment (filippetroviccc) [reply contracts/token/SRC20.sol]: oh, it is in a separate branch, pardon me.

**Pull request closed without merging**